### PR TITLE
Bug fixes in eks cleanup code

### PR DESCRIPTION
**Pull request by rhoboat**
[Test run on terraform-aws-eks](https://app.circleci.com/pipelines/github/gruntwork-io/terraform-aws-eks/1270/workflows/505c8244-9b25-476d-8cbb-8d6006cb65d4/jobs/5936/artifacts) revealed a bug in the logic in our cleanup. This attempts to fix that.

### CODE REVIEW

**Review by yorinasub17 — APPROVED** [review]
This LGTM, assuming my interpretation is correct. Basically the bug was that we were wrapping the error in the stack traced error struct, so the `IsAwsErr` checks were failing?

**Comment by rhoboat** [comment]
The bug was that we were throwing errors that the ENI was NotFound, from within `requestDetach`, when actually, we have a place that catches that error in the calling function:
https://github.com/gruntwork-io/kubergrunt/blob/ca934760a20e1da00ddb7a718da2a27c74f5ce80/eks/cleanup.go#L183-L185

**Comment by rhoboat** [comment]
Sorry, need another approval, couldn't resist improving some comments.

**Comment by yorinasub17** [comment]
> The bug was that we were throwing errors that the ENI was NotFound, from within requestDetach, when actually, we have a place that catches that error in the calling function:

Got it! I think we are saying the same thing.

**Review by yorinasub17 — APPROVED** [review]

**Pull request merged**
